src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -141,18 +102,8 @@ public bool TryGetConfig([NotNullWhen(true)] out RuntimeConfig? runtimeConfig)
     /// <returns>True when runtime config is provided, otherwise false.</returns>
     public bool TryGetLoadedConfig([NotNullWhen(true)] out RuntimeConfig? runtimeConfig)
     {
-        runtimeConfig = _runtimeConfig;
-        return _runtimeConfig is not null;
-    }
-
-    /// <summary>
-    /// Hot Reloads the runtime config when the file watcher
-    /// is active and detects a change to the underlying config file.
-    /// </summary>
-    public void HotReloadConfig()
-    {
-        // _runtimeconfig can not be null in a hot reload scenario
-        ConfigLoader.TryLoadKnownConfig(out _runtimeConfig, replaceEnvVar: true, _runtimeConfig!.DefaultDataSourceName);
+        runtimeConfig = ConfigLoader.RuntimeConfig;
```

Comment:
I don't follow why both `RuntimeConfigProvider` and `RuntimeConfigLoader`(and by definition, any derived classes) maintain the state of `RuntimeConfig`. The responsibilities of these two classes overlap such that it seems incorrect that the `ConfigLoader` is now acting as the 'provider.'